src/bin/driver.rs

```diff
@@ -419,6 +435,81 @@ async fn run_level_last(
     Ok(())
 }
 
+async fn run_aggregate_by_attributes(
+    cfg: &config::Config,
+    mastic: &MasticHistogram,
+    client_0: &CollectorClient,
+    client_1: &CollectorClient,
+    attributes: &[String],
+    num_clients: usize,
+) -> io::Result<()> {
+    for start in (0..num_clients).step_by(cfg.flp_batch_size) {
+        let end = std::cmp::min(num_clients, start + cfg.flp_batch_size);
+        let req = AggregateByAttributesValidateRequest {
+            attributes: attributes.to_vec(),
+            start,
+            end,
+        };
+
+        // For each report, each aggregator evaluates the VIDPF on each of the attributes and returns
+        // the VIDPF proof and its FLP verifier share.
+        let t = Instant::now();
+        let resp_0 = client_0.aggregate_by_attributes_start(long_context(), req.clone());
+        let resp_1 = client_1.aggregate_by_attributes_start(long_context(), req.clone());
+        let (results_0, results_1) = try_join!(resp_0, resp_1).unwrap();
+        assert_eq!(results_0.len(), req.end - req.start);
+        assert_eq!(results_1.len(), req.end - req.start);
+
+        let mut rejected = Vec::with_capacity(req.end - req.start);
+        for (client_index, ((mut verifier, eval_proof_0), (verifier_share_1, eval_proof_1))) in
```

Comment:
```suggestion
        for (client_index, ((mut verifier_share_0, eval_proof_0), (verifier_share_1, eval_proof_1))) in
```
??

Reply:
belowe we acdd verifier_share_0 into verifier, so the result is the actual verifier. I think this makes the variables more clear.